### PR TITLE
Issue 2

### PR DESCRIPTION
**Pull request by sblOWPCKCR**
Issue 2

     + GreenToken: dumb ERC20 contract https://kovan.etherscan.io/address/0x4239B7aB36878a6b28dcE78DdCAF3cb0bd6af40a#code
     + GreenVault https://kovan.etherscan.io/address/0x4f0d1950Aa61258df679c95Cd45bFe8A6499B9b8#code

### CODE REVIEW

File: issue_2/test/TestGreenVault.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      user2: signers[2],
+    }
+  });
+
+  describe("GreenVault", function () {
+    const starting_balance = 1337;
+
+    beforeEach(async function () {
+      this.gtoken = <GreenToken>await deployContract(this.signers.admin,
+        await hre.artifacts.readArtifact("GreenToken"), []);
```
Comment (alcueca) [inline issue_2/test/TestGreenVault.ts:35]: I didn't know I could read the artifacts from the hre! Cool!

File: issue_2/test/TestGreenVault.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      this.gvault = <GreenVault>await deployContract(this.signers.admin,
+        await hre.artifacts.readArtifact("GreenVault"), []);
+
+      for (const user of [this.signers.user1, this.signers.user2]) {
+        await this.gtoken.connect(user).mint(user.address, starting_balance);
+      }
+    });
+
+    it("starts empty", async function () {
+      expect(await this.gvault.connect(this.signers.user1).deposits(this.signers.user1.address, this.gtoken.address))
```
Comment (alcueca) [inline issue_2/test/TestGreenVault.ts]: I would probably use some variables to make the tests more readable:
```
const gvault = this.gvault.connect(this.signers.user1)
const user1 = this.signers.user1.address
```

File: issue_2/test/TestGreenVault.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        .to.be.reverted; // can't check for exact message since it's ERC20-generated and is not prescribed
+    });
+
+    describe("transfers", function () {
+      beforeEach(async function () {
+        await this.gtoken.connect(this.signers.user1).approve(this.gvault.address, starting_balance);
+      });
+      it("change balance", async function () {
+        await expect(() => this.gvault.connect(this.signers.user1).deposit(this.gtoken.address, starting_balance))
+          .to.changeTokenBalances(this.gtoken, [this.signers.user1, this.gvault], [-starting_balance, starting_balance]);
```
Comment (alcueca) [inline issue_2/test/TestGreenVault.ts:71]: I didn't know how to do this either :D

**Review by alcueca — APPROVED** [review]
Minor comment, I learned more from you than you from me this time :D

File: issue_2/test/TestGreenVault.ts (hunk 2 above)
Comment (sblOWPCKCR) [reply issue_2/test/TestGreenVault.ts]: good point. connected gvault to user1 by default

**Pull request merged**